stacks-common/src/util/secp256k1.rs

```diff
@@ -268,6 +270,27 @@ impl Secp256k1PrivateKey {
         }
     }
 
+    /// Create a Secp256k1PrivateKey from seed bytes by repeatedly
+    ///  SHA256 hashing the seed bytes until a private key is found.
+    ///
+    /// If `seed` is a valid private key, it will be returned without hashing.
+    /// The returned private key's compress_public flag will be `true`
+    pub fn from_seed(seed: &[u8]) -> Secp256k1PrivateKey {
```

Comment:
perhaps you should bound the number of loops at some reasonable number so there's no chance of this running forever

Reply:
I'd rather leave as is, because this PR is mostly just refactoring this code. The likelihood of running forever is also exponentially low, and given that a node operator is in control of this seed themselves, they can choose a seed that doesn't loop forever.